fuse_examples/classification/cmmd/runner.py

```diff
@@ -298,57 +250,37 @@ def run_eval(paths: dict, eval_common_params: dict):
     # run
     results = evaluator.eval(ids=None,
-                     data=os.path.join(paths["inference_dir"], eval_common_params["infer_filename"]),
+                     data=os.path.join(cfg.paths.inference_dir, cfg.infer.infer_filename),
                      metrics=metrics,
-                     output_dir=paths['eval_dir'])
+                     output_dir=cfg.paths.eval_dir)
 
     return results
-######################################
-# Run
-######################################
-
 
-if __name__ == "__main__":
-    # allocate gpus
-    if NUM_GPUS == 0:
-        TRAIN_COMMON_PARAMS['manager.train_params']['device'] = 'cpu'
+@hydra.main(config_path="conf", config_name="config")
+def main(cfg : DictConfig) -> None:
+    print(cfg)
     # uncomment if you want to use specific gpus instead of automatically looking for free ones
     force_gpus = None  # [0]
-    FuseUtilsGPU.choose_and_enable_multiple_gpus(NUM_GPUS, force_gpus=force_gpus)
+    FuseUtilsGPU.choose_and_enable_multiple_gpus(cfg.train.manager_train_params.num_gpus, force_gpus=force_gpus)
 
-    RUNNING_MODES = ['train', 'infer', 'eval']  # Options: 'train', 'infer', 'eval'
-    # Path to save model
-    root = ''
-    # Path to the stored CMMD dataset location
+    RUNNING_MODES = ['train', 'infer', 'analyze']  # Options: 'train', 'infer', 'analyze'
+    # Path to the stored dataset location
     # dataset should be download from https://wiki.cancerimagingarchive.net/pages/viewpage.action?pageId=70230508
     # download requires NBIA data retriever https://wiki.cancerimagingarchive.net/display/NBIA/Downloading+TCIA+Images
-    # put on the following in the main folder  - 
+    # put on the folliwing in the main folder  - 
     # 1. CMMD_clinicaldata_revision.csv which is a converted version of CMMD_clinicaldata_revision.xlsx 
     # 2. folder named CMMD which is the downloaded data folder
-    root_data = None #TODO: add path to the data folder
-    assert root_data is not None, "Error: please set root_data, the path to the stored CMMD dataset location"
-    # Name of the experiment
-    experiment = 'model_new/CMMD_classification'
-    # Path to cache data
-    cache_path = 'examples/'
-    # Name of the cached data folder
-    experiment_cache = 'CMMD_'
-    paths = {'data_dir': root_data,
-             'model_dir': os.path.join(root, experiment, 'model_dir_transfer'),
-             'data_misc_dir' : os.path.join(root, 'data_misc'),
-             'force_reset_model_dir': True,
-             # If True will reset model dir automatically - otherwise will prompt 'are you sure' message.
-             'cache_dir': os.path.join(cache_path, experiment_cache + '_cache_dir'),
-             'inference_dir': os.path.join(root, experiment, 'infer_dir'),
-             'eval_dir': os.path.join(root, experiment, 'eval_dir')}
+
     # train
     if 'train' in RUNNING_MODES:
-        run_train(paths=paths, train_common_params=TRAIN_COMMON_PARAMS, reset_cache=False)
+        run_train(cfg)
```

Comment:
is it possible to pass here cfg.paths and cfg.train?
(I want that each function will get just the configuration that it actually using,

Reply:
Is it also possible to convert it to a nested dictionary and keep the code as it used to be?
It will give us the chance to add more definitions here in case it's hard to specify it in hydra format.

Reply:
it's possible